File: Human.cpp
#include "Human.h"
#include "EntityInterface.h"

Human::Human() :Entity()
{
	InstType = HumanType;
	weapon = 0;
	armor = 0;
	name.assign("Default human");
	pouch = 0;
}

Human::Human(int weapon, int armor, char* name, int health) :Entity(health)
{
	InstType = HumanType;
	this->weapon = weapon;
	this->armor = armor;
	this->name.assign(name);
	this->pouch = 0;
}


Human::~Human()
{
	std::cout << "Deleted Human" << std::endl;
}

void Human::Print() const
{
	std::cout << name << " has " << weapon << "damage " << armor << "armor " << MaxHealth << "health " << std::endl;
}

int Human::GetType()
{
	return InstType;
}

void Human::PrintType() const
{
	std::cout << "Im a Human!" << std::endl;
}


int Human::Attack()
{
	std::cout << name << " hits for:" << weapon;
	return weapon;
}

int Human::Defend(int DmgTaken)
{
	if (armor >= DmgTaken)
	{
		std::cout << " but it is too weak to damage you" << std::endl;
		return CurrentHealth;
	}
	CurrentHealth -= (DmgTaken - armor);
	if (CurrentHealth <= 0)
	{
		std::cout << std::endl << name << " died!" << std::endl;
		return 0;
	}
	std::cout << " and " << name << "'s hp now is " << CurrentHealth << std::endl;
	return CurrentHealth;
}


void Human::AddCoins(int Amount)
{
	if (pouch == 0 && Amount <= 0)
	{
		return;
	}
	else
	{
		pouch += Amount;
	}
}

int Human::GetCoins()
{
	return pouch;
}

void Human::UseItem(int ItemID, int Effect)
{
	if (ItemID == 1)
	{
		CurrentHealth += Effect;
		if (CurrentHealth  > MaxHealth)
		{
			CurrentHealth = MaxHealth;
			return;

		}
	}
	else if (ItemID == 3)
	{
		MaxHealth += Effect;
		return;
	}
}

File: Java version/Human.java



public class Human extends Entity {
    private int Weapon;
    private int Armor;
    private int Pouch;
    private String Name;

    public Human(int Weapon, int Armor, String Name, int MaxHealth) {
        super(Type.Human, MaxHealth);
        this.Weapon = Weapon;
        this.Armor = Armor;
        this.Pouch = 0;
        this.Name = Name;
    }


    //text output methods
    public void PrintType() {
        System.out.println("human");
    }

    public void Print() {
        System.out.println(Name + " has " + Weapon + " damage "
         + Armor + " armor " + CurrentHealth + "/" +  MaxHealth + " health");
    }


    //getters and setters
    public Type GetType() {
        return InstType;
    }

    public int GetCoins() {
        return Pouch;
    }

    public void AddCoins(int Amount) {
        if (Pouch == 0 && Amount <= 0) {
            return;
        }
        else {
            Pouch += Amount;
        }
    }


    //gameplay methods
    public int Attack() {
        System.out.print(Name + " hits for: " + Weapon);
        return Weapon;
    }

    public int Defend(int Dmgtaken) {
        if (Armor >= Dmgtaken) {
            System.out.print(" but it is too weak to damage you!");
            System.out.println("");
            return CurrentHealth;
        }
        CurrentHealth -= (Dmgtaken - Armor);
        if (CurrentHealth <= 0) {
            System.out.println(Name + " died!");
            return 0;
        }
        System.out.println(" and " + Name + "'s hp is now " + CurrentHealth);
        return CurrentHealth;
    }

    public void UseItem(ItemType Type, int Effect) {
        if(Type == ItemType.HealingItem) {
            CurrentHealth += Effect;
            if (CurrentHealth > MaxHealth) {
                CurrentHealth = MaxHealth;
                return;
            }
        }
        else if (Type == ItemType.MaxHPIncreaseItem) {
            MaxHealth += Effect;
            return;
        }
    }

}